conformance/lang/expressions/literals/floating_point_literal.balt

```diff
@@ -0,0 +1,6092 @@
+    a = 0x3eA3p.34; // @error invalid HexFloatingPointLiteral due to invalid sign in HexExponent
+
+    a = 0x3eA3p+ 34; // @error invalid HexFloatingPointLiteral due to unnecessary space
+   
+    a = 0x3eA3p+34f; // @error invalid HexFloatingPointLiteral due to unnecessary suffix
+
+    a = 0x3eA3p+34F; // @error invalid HexFloatingPointLiteral due to unnecessary suffix
+
+    a = 0x3eA3p+34D; // @error invalid HexFloatingPointLiteral due to unnecessary suffix
+
+    a = 0x3eA3p+34d; // @error invalid HexFloatingPointLiteral due to unnecessary suffix
+
+    a = 0x3eA3H-34; // @error invalid HexFloatingPointLiteral
+
+    a = 8x3eA3p67; // @error invalid HexFloatingPointLiteral due to invalid HexIndicator
+
+    a = 0Z3eA3p67; // @error invalid HexFloatingPointLiteral due to invalid HexIndicator
+    
+    a = 0x3eA3 .4Eb2; // @error invalid HexFloatingPointLiteral due to unnecessary space
+
+    a = 0x3eA3. 4Eb2; // @error invalid HexFloatingPointLiteral due to unnecessary space
+
+    a = 0x3eZ3.4Eb2p45; // @error invalid HexFloatingPointLiteral due to invalid HexDigit
+
+    a = 0x3e3.4EbZ2p45; // @error invalid HexFloatingPointLiteral due to invalid HexDigit
+
+    a = 0x3e3z.4EbZ2p45; // @error invalid HexFloatingPointLiteral due to invalid HexDigit
+    
+    a = 0x3e3.4Eb2p45Z; // @error invalid HexFloatingPointLiteral due to invalid HexExponent
+
+    a = 0x3e3..4Eb2p45; // @error invalid HexFloatingPointLiteral
+
+    a = 0x3e3=4Eb2p45; // @error invalid HexFloatingPointLiteral
+
+    a = .0x3e3; // @error invalid HexFloatingPointLiteral
+
+    a = 0x3e3 .p23; // @error invalid HexFloatingPointLiteral due to unnecessary space
+
+    a = 0x3e3. p23; // @error invalid HexFloatingPointLiteral due to unnecessary space
+
+    a = 3eA3.4Eb2p0x+45; // @error invalid HexFloatingPointLiteral
+
+    a = 0x3eA3.4Eb2pA+45; // @error invalid HexFloatingPointLiteral
+
+    a = 0x..4Eb2P-45; // @error invalid HexFloatingPointLiteral
+
+    a = 0x4E,b2P-45; // @error invalid HexFloatingPointLiteral
+
+    a = 0x4E:b2P-45; // @error invalid HexFloatingPointLiteral
+
+    a = 0x. 4Eb2P-45; // @error invalid HexFloatingPointLiteral due to unnecessary space
+
+    a = 0x .4Eb2P-45; // @error invalid HexFloatingPointLiteral due to unnecessary space
+}
+
+Test-Case: error
+Description: Test invalid hex floating point literal
+Labels: decimal, float, HexFloatingPointLiteral
+
+function init() {
+    float|decimal a = .99+1742; // @error invalid DecimalFloatingPointNumber due to missing ExponentIndicator
+    a = e+174; // @error invalid DecimalFloatingPointNumber due to the absence of DecimalNumber or DottedDecimalNumber
+    a = e174; // @error invalid DecimalFloatingPointNumber due to the absence of DecimalNumber or DottedDecimalNumber
+}
+
+Test-Case: error
+Description: Test invalid hex floating point literal
+Fail-Issue: ballerina-lang #32750 #32840
+Labels: decimal, float, HexFloatingPointLiteral
+
+function init() {
+    float|decimal a = 0x 3eA3p34;
```

Comment:
Move the failing cases separately and add the expected error behaviour

Reply:
[808331d](https://github.com/ballerina-platform/ballerina-spec/pull/947/commits/808331da7e85b86e4b950f73bcddde9b67a754dc)